logstash-core/src/test/java/org/logstash/FileLockFactoryTest.java

```diff
@@ -88,4 +90,52 @@ public void ReleaseUnobtainedLock() throws IOException {
         FileLockFactory.getDefault().releaseLock(lock);
         FileLockFactory.getDefault().releaseLock(lock);
     }
+
+    @Test
+    public void crossJvmObtainLockOnLocked() throws IOException {
+        // Timeout (in ms).
+        long timeout = 30000;
+        String fs = System.getProperty("file.separator");
+        Process p = null;
+        String lockFile = ".testCrossJvm";
+        FileLock lock = null;
+
+        // Build the command to spawn a children JVM.
+        String[] cmd = {
+            System.getProperty("java.home") + fs + "bin" + fs + "java",
+            "-cp", System.getProperty("java.class.path"),
+            "org.logstash.FileLockFactoryMain",
+            lockDir, lockFile
+        };
+
+        try {
+            // Start the children program that will lock the file.
+            ProcessBuilder pb = new ProcessBuilder(cmd);
+            p = pb.start();
+            InputStream is = p.getInputStream();
+            long endTime = System.currentTimeMillis() + timeout;
+            long now;
+            do {
+                // Wait on children program to write something to stdout.
+                now = System.currentTimeMillis();
+            } while (is.available() == 0 && now < endTime);
```

Comment:
@wainersm wait, are you sure about this? `finally {}` should run on even `ThreadDeath` being thrown. But doesn't JUnit simply interrupt and try to join the thread?
In that case `finally {}` will be run for sure or am I missing something? :)